File: Player_ObjC/PlayerDemoObjC/Player/PlayerPreview.m
//
//  PlayerPreview.m
//  HiARSDKComponent
//
//  Created by JT Ma on 18/09/2017.
//  Copyright © 2017 MaJiangtao<majt@hiscene.com>. All rights reserved.
//

#import "PlayerPreview.h"

@implementation PlayerPreview

+ (Class)layerClass {
    return [AVPlayerLayer class];
}

- (AVPlayerLayer *)playerLayer {
    return (AVPlayerLayer *)self.layer;
}

- (AVPlayer *)player {
    return self.playerLayer.player;
}

- (void)setPlayer:(AVPlayer *)player {
    if (!self.playerLayer.player) {
        self.playerLayer.player = player;
    }
}

@end

File: OYE/Player/PlayerPreview.swift
//
//  PlayerPreview.swift
//  OYE
//
//  Created by JT Ma on 30/09/2017.
//  Copyright © 2017 JiangtaoMa<majt@hiscene.com>. All rights reserved.
//

import UIKit
import AVFoundation

public class PlayerPreview: UIView {
    
    public var playerLayer: AVPlayerLayer {
        return (layer as? AVPlayerLayer)!
    }
    
    public var player: AVPlayer? {
        get {
            return playerLayer.player
        }
        set {
            playerLayer.player = newValue
        }
    }
    
    public override class var layerClass: AnyClass {
        return AVPlayerLayer.self
    }
}
